(err, res) => {
          if (err) return reject(err)
          resolve(res)
        }